src/main/java/org/example/hw20/jdbc/util/ConnectionManager.java

```diff
@@ -0,0 +1,83 @@
+package org.example.hw20.jdbc.util;
+
+import java.lang.reflect.Proxy;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+public class ConnectionManager {
+    private static final String PASSWORD_KEY = "db.password";
+    private static final String USERNAME_KEY = "db.username";
+    private static final String URL_KEY = "db.url";
+    private static final String POOL_SIZE_KEY = "db.pool.size";
+    private static final Integer DEFAULT_POOL_SIZE = 10;
+    private static BlockingQueue<Connection> pool;
+    private static List<Connection> sourceConnections;
+
+    static {
+        loadDriver();
+        initConnectionPool();
+    }
+
+    private ConnectionManager() {
+    }
+
+    private static void initConnectionPool() {
+        var poolSize = PropertiesUtil.get(POOL_SIZE_KEY);
+        var size = poolSize == null ? DEFAULT_POOL_SIZE : Integer.parseInt(poolSize);
+        pool = new ArrayBlockingQueue<>(size);
+        sourceConnections = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            var connection = open();
+            var proxyConnection = (Connection)
+                    Proxy.newProxyInstance(ConnectionManager.class.getClassLoader(), new Class[]{Connection.class},
+                            (proxy, method, args) -> method.getName().equals("close")
+                                    ? pool.add((Connection) proxy)
+                                    : method.invoke(connection, args));
+            pool.add(proxyConnection);
+            sourceConnections.add(connection);
+        }

```

Comment:
Like from me)
If you understand what's happening here, of course. But nice that you tried this